### ISSUE TITLE
integrated.py: empty-result paths disagree with the normal paths (hardcoded 128 dims, LAF rank, negative confidence)

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4065]
### Describe the bug

Three related inconsistencies in `kornia/feature/integrated.py`, all on the "nothing was found" paths, where the empty result does not have the same shape or semantics as the non-empty one.

#### 1. `get_laf_descriptors` hardcodes a descriptor width of 128 when there are no keypoints

https://github.com/kornia/kornia/blob/main/kornia/feature/integrated.py#L75

```python
if lafs.shape[1] == 0:
    warnings.warn(...)
    return torch.empty(lafs.shape[0], lafs.shape[1], 128, dtype=lafs.dtype, device=lafs.device)
```

The width is taken from the descriptor module on the normal path but hardcoded to 128 here, so the returned descriptor dimension depends on whether any keypoints were detected.

```python
import torch, warnings
from kornia.feature import get_laf_descriptors, MKDDescriptor
from kornia.feature.laf import laf_from_center_scale_ori

img = torch.rand(1, 1, 64, 64)
laf_ok = laf_from_center_scale_ori(torch.tensor([[[32., 32.]]]), torch.full((1, 1, 1, 1), 8.0))
laf_empty = torch.zeros(1, 0, 2, 3)

for dims in (64, 128, 238):
    desc = MKDDescriptor(patch_size=32, output_dims=dims)
    with warnings.catch_warnings():
        warnings.simplefilter("ignore")
        print(f"output_dims={dims:4d}  non-empty={get_laf_descriptors(img, laf_ok, desc, 32).shape[-1]:4d}"
              f"  empty={get_laf_descriptors(img, laf_empty, desc, 32).shape[-1]:4d}")
```

```
output_dims=  64  non-empty=  64  empty= 128
output_dims= 128  non-empty= 128  empty= 128
output_dims= 238  non-empty= 238  empty= 128
```

The empty tensor also takes `dtype=lafs.dtype` rather than the descriptor's output dtype.

#### 2. `LocalFeatureMatcher` returns a different LAF rank when there are no matches

`no_match_output` returns `lafs0` / `lafs1` with shape `(0, 0, 2, 3)`:

https://github.com/kornia/kornia/blob/main/kornia/feature/integrated.py#L407

while the success path returns `(1, N, 2, 3)`:

https://github.com/kornia/kornia/blob/main/kornia/feature/integrated.py#L483

The batch dimension is 0 in one case and 1 in the other, so caller code that does `out["lafs0"][0]` works on a successful match and raises on an empty one. The sibling `keypoints0` is consistent between the two paths (`(0, 2)` vs `(-1, 2)`); only the LAFs disagree.

#### 3. `confidence` can be negative

https://github.com/kornia/kornia/blob/main/kornia/feature/integrated.py#L469

```python
out_confidence.append(1.0 - dists)
```

`forward` documents `confidence` as a *"confidence score [0, 1]"*. That holds when the matcher returns a ratio (`snn`, `smnn`, `fginn`), but `match_nn` / `match_mnn` return raw L2 descriptor distances, which reach 2.0 for unit-norm descriptors — so `DescriptorMatcher("nn")` yields confidences down to −1.

### Expected behavior

1. Empty-path descriptor width should come from the descriptor module, and the dtype should match the non-empty path.
2. `no_match_output` should return `lafs0` / `lafs1` with the same rank and batch dimension as the success path.
3. Either normalise the distance before converting it to a confidence, or document that `confidence` is only in `[0, 1]` for ratio-based matchers.

### Additional context

`integrated.py` sits at 51% line coverage, the lowest of the hand-written (non-vendored) files in `kornia/feature`; items 2 and 3 are on lines that no test exercises.

Found during an audit of `kornia.feature`.

Posted on behalf of @ducha-aiki by Claude (Opus 5).
